staging/src/k8s.io/apiextensions-apiserver/test/integration/finalization_test.go

```diff
@@ -89,3 +89,68 @@ func TestFinalization(t *testing.T) {
 	require.Error(t, err)
 	require.True(t, errors.IsNotFound(err), "%#v", err)
 }
+
+func TestFinalizationAndDeletion(t *testing.T) {
+	stopCh, apiExtensionClient, clientPool, err := testserver.StartDefaultServerWithClients()
+	require.NoError(t, err)
+	defer close(stopCh)
+
+	// Create a CRD.
+	noxuDefinition := testserver.NewNoxuCustomResourceDefinition(apiextensionsv1beta1.ClusterScoped)
+	noxuVersionClient, err := testserver.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, clientPool)
+	require.NoError(t, err)
+
+	// Create a CR with a finalizer.
+	ns := "not-the-default"
+	name := "foo123"
+	noxuResourceClient := NewNamespacedCustomResourceClient(ns, noxuVersionClient, noxuDefinition)
+
+	instance := testserver.NewNoxuInstance(ns, name)
+	instance.SetFinalizers([]string{"noxu.example.com/finalizer"})
+	createdNoxuInstance, err := instantiateCustomResource(t, instance, noxuResourceClient, noxuDefinition)
+	require.NoError(t, err)
+
+	// Delete a CR. Because there's a finalizer, it will not get deleted now.
+	uid := createdNoxuInstance.GetUID()
+	err = noxuResourceClient.Delete(name, &metav1.DeleteOptions{
+		Preconditions: &metav1.Preconditions{
+			UID: &uid,
+		},
+	})
+	require.NoError(t, err)
+
+	// Check is the CR scheduled for deletion.
+	gottenNoxuInstance, err := noxuResourceClient.Get(name, metav1.GetOptions{})
+	require.NoError(t, err)
+	require.NotNil(t, gottenNoxuInstance.GetDeletionTimestamp())
+
+	// Delete the CRD.
+	testserver.DeleteCustomResourceDefinition(noxuDefinition, apiExtensionClient)
+
+	// Check is CR still there after the CRD deletion.
+	gottenNoxuInstance, err = noxuResourceClient.Get(name, metav1.GetOptions{})
+	require.NoError(t, err)
+
+	// Update the CR to remove the finalizer.
+	for {
+		gottenNoxuInstance.SetFinalizers(nil)
+		_, err = noxuResourceClient.Update(gottenNoxuInstance)
+		if err == nil {
+			break
+		}
+		if !errors.IsConflict(err) {
+			require.NoError(t, err) // Fail on unexpected error
+		}
+		gottenNoxuInstance, err = noxuResourceClient.Get(name, metav1.GetOptions{})
+		require.NoError(t, err)
+	}
+
+	// Verify the CR is gone.
+	// It should return the NonFound error.
+	gottenNoxuInstance, err = noxuResourceClient.Get(name, metav1.GetOptions{})
+	require.Error(t, err)
```

Comment:
For checking if the error is `NotFound`, you can use `apierrors.IsNotFound(err)` where `apierrors` is `"k8s.io/apimachinery/pkg/api/errors"`